### PR TITLE
Additional parts

### LINKED ISSUE #79: Additional parts with color and size mapping

### PR DESCRIPTION
**Pull request by KaviyaPeriyasamy**
Fixes #79
The creation of additional parts with color and size mapping, adding filters for item field in respective doctypes, creation of item group for the intermediate product.

### CODE REVIEW

File: apparelo/apparelo/doctype/stitching/stitching.js [hunk 1]
```diff
@@ -43,5 +43,94 @@ frappe.ui.form.on('Stitching', {
...
+			};
+		});
+		frm.set_query("items", function() {
+			return {
+				filters: {
+					"item_group":"Raw Material"
+				}
+			};
+		});
+		frm.set_query("additional_items", function() {
```
Comment (karthikeyan5) [inline apparelo/apparelo/doctype/stitching/stitching.js]: is it possible send these filters to a single file and include here?

File: apparelo/apparelo/doctype/stitching/stitching.js [hunk 2]
```diff
@@ -43,5 +43,94 @@ frappe.ui.form.on('Stitching', {
...
+			return {
+				filters: {
+					"item_group":"Raw Material",
```
Comment (karthikeyan5) [inline apparelo/apparelo/doctype/stitching/stitching.js]: add `Has Variants` filter also... here and all other related places

File: apparelo/apparelo/doctype/stitching/stitching.json [hunk 3]
```diff
@@ -16,8 +16,26 @@
   "section_break_7",
   "colour_mappings",
   "parts_per_piece",
-  "additional_parts",
   "additional_information",
+  "section_break_12",
+  "enable_additional_parts",
+  "section_break_24",
+  "items",
```
Comment (karthikeyan5) [inline apparelo/apparelo/doctype/stitching/stitching.json]: can we rename this to a more related name? something like `size_additional_items`?

File: apparelo/apparelo/doctype/stitching/stitching.json [hunk 4]
```diff
@@ -16,8 +16,26 @@
...
+  "get_piece_color",
+  "column_break_15",
+  "piece_colors",
+  "is_part_color_same_as_piece_color",
+  "section_break_18",
+  "additional_parts_colour",
+  "section_break_23",
+  "additional_items",
```
Comment (karthikeyan5) [inline apparelo/apparelo/doctype/stitching/stitching.json]: same as above... `color_additional_item`?

Comment (karthikeyan5) [reply apparelo/apparelo/doctype/stitching/stitching.json]: also do the same for all other doctypes....

File: apparelo/apparelo/doctype/stitching/stitching.py [hunk 5]
```diff
@@ -116,9 +117,12 @@ def create_boms(self, input_item_names, variants,attribute_set,item_size,colour,
 									if size.upper() in input_item  and size.upper() in variant and colour_mapping.piece_colour.upper() in variant and colour_mapping.part.upper() in input_item and colour_mapping.part_colour.upper() in input_item:
 										if piece_count.part==colour_mapping.part:
 											item_list.append({"item_code": input_item,"qty":piece_count.qty ,"uom": "Nos"})
-			if not self.additional_parts==[]:
-				for item in self.additional_parts:
-						item_list.append({"item_code": item.item,"uom": "Nos","qty":item.qty})
+			if self.enable_additional_parts:
+				matched_part=matching_additional_part(additional_parts,self.additional_parts_colour,self.additional_parts_size,self.additional_parts,variant)
+				for additional_ in self.additional_parts:
```
Comment (karthikeyan5) [inline apparelo/apparelo/doctype/stitching/stitching.py]: rename to `additional_part`

**Review by karthikeyan5 — CHANGES_REQUESTED** [review]

File: apparelo/apparelo/doctype/stitching/stitching.py (hunk 5 above)
Comment (KaviyaPeriyasamy) [reply apparelo/apparelo/doctype/stitching/stitching.py]: This commit resolve all the changes 749652cfbf24ee57d93d639b3057e167e86064bb

**Review by karthikeyan5 — CHANGES_REQUESTED** [review]
issues found: 
- [x] `Get Piece Color` and `Get Piece Size` Button not working... getting error following error
<details>
<summary>Traceback</summary>

```Python traceback
Traceback (most recent call last):
  File "/home/karthi/frappe/benches/dev-bench/apps/frappe/frappe/app.py", line 60, in application
    response = frappe.api.handle()
  File "/home/karthi/frappe/benches/dev-bench/apps/frappe/frappe/api.py", line 55, in handle
    return frappe.handler.handle()
  File "/home/karthi/frappe/benches/dev-bench/apps/frappe/frappe/handler.py", line 24, in handle
    data = execute_cmd(cmd)
  File "/home/karthi/frappe/benches/dev-bench/apps/frappe/frappe/handler.py", line 63, in execute_cmd
    return frappe.call(method, **frappe.form_dict)
  File "/home/karthi/frappe/benches/dev-bench/apps/frappe/frappe/__init__.py", line 1059, in call
    return fn(*args, **newargs)
  File "/home/karthi/frappe/benches/dev-bench/apps/apparelo/apparelo/apparelo/doctype/stitching/stitching.py", line 225, in get_additional_item_piece_colour
    for item in doc.get('items'):
TypeError: 'NoneType' object is not iterable
```
</details>

- [x] submit IPD is not working getting the following error
<details>
<summary>Traceback</summary>

```Python traceback
Traceback (most recent call last):
  File "/home/karthi/frappe/benches/dev-bench/apps/frappe/frappe/desk/form/save.py", line 19, in savedocs
    doc.submit()
  File "/home/karthi/frappe/benches/dev-bench/apps/frappe/frappe/model/document.py", line 875, in submit
    self._submit()
  File "/home/karthi/frappe/benches/dev-bench/apps/frappe/frappe/model/document.py", line 864, in _submit
    self.save()
  File "/home/karthi/frappe/benches/dev-bench/apps/frappe/frappe/model/document.py", line 277, in save
    return self._save(*args, **kwargs)
  File "/home/karthi/frappe/benches/dev-bench/apps/frappe/frappe/model/document.py", line 330, in _save
    self.run_post_save_methods()
  File "/home/karthi/frappe/benches/dev-bench/apps/frappe/frappe/model/document.py", line 941, in run_post_save_methods
    self.run_method("on_submit")
  File "/home/karthi/frappe/benches/dev-bench/apps/frappe/frappe/model/document.py", line 803, in run_method
    out = Document.hook(fn)(self, *args, **kwargs)
  File "/home/karthi/frappe/benches/dev-bench/apps/frappe/frappe/model/document.py", line 1090, in composer
    return composed(self, method, *args, **kwargs)
  File "/home/karthi/frappe/benches/dev-bench/apps/frappe/frappe/model/document.py", line 1073, in runner
    add_to_return_value(self, fn(self, *args, **kwargs))
  File "/home/karthi/frappe/benches/dev-bench/apps/frappe/frappe/model/document.py", line 797, in <lambda>
    fn = lambda self, *args, **kwargs: getattr(self, method)(*args, **kwargs)
  File "/home/karthi/frappe/benches/dev-bench/apps/apparelo/apparelo/apparelo/doctype/item_production_detail/item_production_detail.py", line 18, in on_submit
    ipd_list=self.create_process_details()
  File "/home/karthi/frappe/benches/dev-bench/apps/apparelo/apparelo/apparelo/doctype/item_production_detail/item_production_detail.py", line 219, in create_process_details
    bom,new_variants=cutting_doc.create_boms(input_items, variants, attribute_set,item_size,colour,piece_count)
  File "/home/karthi/frappe/benches/dev-bench/apps/apparelo/apparelo/apparelo/doctype/cutting/cutting.py", line 119, in create_boms
    bom_doc=frappe.get_doc("BOM",bom_)
  File "/home/karthi/frappe/benches/dev-bench/apps/frappe/frappe/__init__.py", line 747, in get_doc
    doc = frappe.model.document.get_doc(*args, **kwargs)
  File "/home/karthi/frappe/benches/dev-bench/apps/frappe/frappe/model/document.py", line 71, in get_doc
    return controller(*args, **kwargs)
  File "/home/karthi/frappe/benches/dev-bench/apps/frappe/frappe/website/website_generator.py", line 16, in __init__
    super(WebsiteGenerator, self).__init__(*args, **kwargs)
  File "/home/karthi/frappe/benches/dev-bench/apps/frappe/frappe/model/document.py", line 106, in __init__
    self.load_from_db()
  File "/home/karthi/frappe/benches/dev-bench/apps/frappe/frappe/model/document.py", line 149, in load_from_db
    frappe.throw(_("{0} {1} not found").format(_(self.doctype), self.name), frappe.DoesNotExistError)
  File "/home/karthi/frappe/benches/dev-bench/apps/frappe/frappe/__init__.py", line 372, in throw
    msgprint(msg, raise_exception=exc, title=title, indicator='red')
  File "/home/karthi/frappe/benches/dev-bench/apps/frappe/frappe/__init__.py", line 358, in msgprint
    _raise_exception()
  File "/home/karthi/frappe/benches/dev-bench/apps/frappe/frappe/__init__.py", line 315, in _raise_exception
    raise raise_exception(msg)
frappe.exceptions.DoesNotExistError: BOM None not found
```
</details>

**Pull request merged**